src/camera/Camera.h

```diff
@@ -0,0 +1,295 @@
+	bool open(int camera_id, CameraParams intrinsic_params = CameraParams(),
+			  cv::Mat extrinsic_params = cv::Mat());
+	/**
+	   @brief Constructs a Camera that will open the given file for input and
+	   have the given name and description.
+
+	   @param filename The file to open and read a video feed from. This may be
+	   the name of a video file, or a URI of a video stream. This will be passed
+	   to the underlying OpenCV VideoCapture object, so anything supported by
+	   VideoCapture is supported here.
+
+	   @param name The name of the camera. This should ideally be unique, but
+	   isn't enforced at this time.
+
+	   @param description An optional description of the camera.
+	 */
+	Camera(std::string filename, std::string name, std::string description = "",
+		   CameraParams intrinsic_params = CameraParams(),
+		   cv::Mat extrinsic_params = cv::Mat());
+	/**
+	   @brief Constructs a Camera that will open the camera with the given ID
+	   and have the given name and description.
+
+	   @param filename The file to open and read a video feed from. This may be
+	   the name of a video file, or a URI of a video stream. This will be passed
+	   to the underlying OpenCV VideoCapture object, so anything supported by
+	   VideoCapture is supported here.
+
+	   @param name The name of the camera. This should ideally be unique, but
+	   isn't enforced at this time.
+
+	   @param description An optional description of the camera.
+	*/
+	Camera(int camera_id, std::string name, std::string description = "",
+		   CameraParams intrinsic_params = CameraParams(),
+		   cv::Mat extrinsic_params = cv::Mat());
+	/**
+	   @brief Copy constructor.
+
+	   A Camera object created through this constructor will share access to the
+	   underlying camera with the Camera object it is copying. When all Camera
+	   objects sharing access are destroyed, the camera will be closed.
+	 */
+	Camera(const Camera &other);
+
+	/**
+	   @brief Opens the camera using the given configuration file.
+
+	   @param filename The path to the configuration file to open and
+	   read. Configuration file should be formatted as described in @ref
+	   cameraconfig.
+
+	   @throws invalid_camera_config If the configuration is invalid for any
+	   reason.
+	*/
+	bool openFromConfigFile(std::string filename);
+
+	/**
+	   @brief Returns true if the camera is open.
+	 */
+	bool isOpen() const;
+	/**
+	   @brief Returns true if the Camera object has a frame that is different
+	   from the last one the client retrieved.
+
+	   @param last_frame_num The frame number that was returned when retrieving
+	   a frame.
+
+	   @returns true if there is a new frame (i.e. the current frame number is
+	   different than the given one) and false if not.
+	 */
+	bool hasNext(uint32_t last_frame_num) const;
+	/**
+	   @brief Retrieves the next frame.
+
+	   @param[out] frame An output parameter for the frame. Whatever is passed
+	   in will be overwritten so you do not need to worry about the passed-in
```

Comment:
I was briefly confused by this wording. Maybe instead of "Whatever is passed in will be overwritten" just say "Everything will be overwritten"

Reply:
Agreed, seems better.